api/internal/core/entity/format.go

```diff
@@ -17,17 +17,23 @@
 package entity
 
 import (
-	"net"
+	"errors"
 	"strconv"
+	"strings"
 
 	"github.com/apisix/manager-api/internal/log"
 )
 
 func mapKV2Node(key string, val float64) (*Node, error) {
-	host, port, err := net.SplitHostPort(key)
-	if err != nil {
-		log.Errorf("split host port fail: %s", err)
-		return nil, err
+	hp := strings.Split(key, ":")
+	host := hp[0]
+	//  according to APISIX upstream nodes policy, port is optional
+	port := "0"
```

Comment:
Is it appropriate to use port 0 by default? Should port 80 be used by default? 🤔

Reply:
https://github.com/apache/apisix-dashboard/blob/92f46d9d2fa8438659131eedecf9cab4c87b0f9d/api/internal/core/entity/entity.go#L99-L105

We cannot give default values to port, the value of 0 is assigned because the zero value is ignored during `json.Marshal`

Reply:
![image](https://user-images.githubusercontent.com/8078418/168734898-092514d5-8de6-4bd5-89fd-f0eda22b08ba.png)

When there is no port in host, port `0` will be used by default, which I think is wrong. For HTTP it should be `80` and for HTTPS it should be `443`, so I think it should be set to 80 by default.

Reply:
Port is not required, when using https and not configured port it will cause problems, if we do not configure the port there is no such problem

If 80 is given, you need to determine the schema type, when it is https and then change the port default back to 443. I think the first one is more convenient, what do you think?

Reply:
So to put it another way, if the port is 0, the JSON will not contain the port field when it is encoded in JSON? The `port` is required.

![image](https://user-images.githubusercontent.com/8078418/168751717-62b6e9ed-06ad-4a7c-8b29-97b511dabc7b.png)

![image](https://user-images.githubusercontent.com/8078418/168751989-92deb7a2-5654-4bf1-847f-3bbe6d0e2dbf.png)